apicast/src/proxy.lua

```diff
@@ -309,6 +310,19 @@ function _M:access(service)
 
   local credentials, err = service:extract_credentials()
 
+  if backend_version == 'oauth' and keycloak then
+    local k = oauth.new()
+    local jwt = k.parse_and_verify_token(credentials.access_token, k.config.public_key)
+
+    if jwt then
```

Comment:
all this can be moved to the keycloak module and just accept `credentials` as an argument